khiops/utils/dataset.py

```
@@ -29,12 +32,268 @@
+                raise TypeError(
+                    type_error_message(
+                        f"'{table_name}' table's key column name",
+                        column_name,
+                        str,
+                    )
+                )
+
+
+def _check_multitable_spec(ds_spec):
+    assert len(ds_spec) > 1
+    # Check the main table
+    if "main_table" not in ds_spec:
+        raise ValueError(
+            "'main_table' entry must be specified for multi-table datasets"
+        )
+    if not isinstance(ds_spec["main_table"], str):
+        raise TypeError(
+            type_error_message("'main_table' entry", ds_spec["main_table"], str)
+        )
+    if ds_spec["main_table"] not in ds_spec["tables"]:
+        raise ValueError(
+            f"A table entry with the main table's name ('{ds_spec['main_table']}') "
+            f"must be present in the 'tables' dictionary"
+        )
+
+    # Check that all tables have non-None keys
+    for table_name, (_, table_key) in ds_spec["tables"].items():
+        if table_key is None:
+            table_type = "main" if ds_spec["main_table"] == table_name else "secondary"
+            raise ValueError(
+                f"key of {table_type} table '{table_name}' is 'None': "
+                "table keys must be specified in multi-table datasets"
+            )
+
+    # If the 'relations' entry exists check it
+    if "relations" in ds_spec:
+        relations_spec = ds_spec["relations"]
+    # Otherwise build a star schema relations spec and check it
+    else:
+        relations_spec = [
+            (ds_spec["main_table"], table)
+            for table in ds_spec["tables"].keys()
+            if table != ds_spec["main_table"]
+        ]
+    _check_relations_entry(ds_spec["main_table"], ds_spec["tables"], relations_spec)
+
+
+def _check_relations_entry(main_table_name, tables_spec, relations_spec):
+    # Check the types and size of the relation entries
+    if not is_list_like(relations_spec):
```

Why is this not an `assert` as the error could only be a programmatic one, because this is a module-private function?

---

Again, because it is a check method and we want to propagate the error to the user.